clarifai/utils/model_train.py

```diff
@@ -0,0 +1,149 @@
+  train_dict["output_info"] = dict()
+
+  train_dict["train_info"]['params'] = Struct()
+  if 'custom_config' in params_dict['train_params'].keys():
+    # Open and read the Python file
+    with open(params_dict['train_params']['custom_config'], 'r') as python_file:
+      custom_config = python_file.read()
+    params_dict['train_params']['custom_config'] = custom_config
+
+  if 'base_embed_model' in params_dict['train_params'].keys():
+    train_dict["input_info"] = dict()
+    train_dict["input_info"]['base_embed_model'] = params_dict['train_params']['base_embed_model']
+    train_dict['input_info'] = resources_pb2.InputInfo(**train_dict['input_info'])
+    del params_dict['train_params']['base_embed_model']
+
+  train_dict["train_info"]['params'].update(params_dict["train_params"])
+  train_dict["train_info"]['params']['dataset_id'] = params_dict['dataset_id']
+  train_dict["train_info"]['params']['dataset_version_id'] = params_dict['dataset_version_id']
+  train_dict['train_info'] = resources_pb2.TrainInfo(**train_dict['train_info'])
+
+  if 'concepts' in params_dict.keys():
+    train_dict["output_info"]['data'] = resources_pb2.Data(
+        concepts=[resources_pb2.Concept(id=concept_id) for concept_id in params_dict["concepts"]])
+  if 'inference_params' in params_dict.keys():
+    train_dict["output_info"]['params'] = Struct()
+    train_dict['output_info']['params'].update(params_dict["inference_params"])
+  train_dict['output_info'] = resources_pb2.OutputInfo(**train_dict['output_info'])
+
+  return train_dict
+
+
+def response_to_param_info(response: MultiModelTypeResponse,
+                           model_type_id: str,
+                           param: str,
+                           template: str = None) -> Dict[str, Any]:
+  """Converts the response from the API to a dictionary of model param info for the given model type id."""
+  dict_response = MessageToDict(response)
+  for model_type in dict_response['modelTypes']:
+    if model_type['id'] == model_type_id:
+      for modeltypefield in model_type['modelTypeFields']:
+        if modeltypefield['path'].split('.')[-1] == param:
+          if param == 'template':
+            del modeltypefield['placeholder']
+            del modeltypefield['modelTypeEnumOptions']
+            return modeltypefield
+          modeltypefield['param'] = modeltypefield.pop('path').split('.')[-1]
+          del modeltypefield['placeholder']
+          return modeltypefield
+        if modeltypefield['path'].split('.')[-1] == "template":
+          for modeltypeenum in modeltypefield['modelTypeEnumOptions']:
+            if modeltypeenum['id'] == template:
+              for modeltypeenumfield in modeltypeenum['modelTypeFields']:
+                if modeltypeenumfield['path'].split('.')[-1] == param:
+                  modeltypeenumfield['param'] = modeltypeenumfield.pop('path').split('.')[-1]
+                  del modeltypeenumfield['placeholder']
+                  return modeltypeenumfield
```

Comment:
Some comments could be helpful to maintain in future

Reply:
Sure. We can add comments as `parsing the API response Multimodeltypes.` But every loop detail, not sure. Thats why kept the iter variable names clearly.